cmd/microcloud/status.go

```diff
@@ -0,0 +1,444 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/canonical/microcluster/v2/microcluster"
+	microTypes "github.com/canonical/microcluster/v2/rest/types"
+	"github.com/charmbracelet/lipgloss"
+	"github.com/charmbracelet/lipgloss/table"
+	"github.com/spf13/cobra"
+
+	"github.com/canonical/microcloud/microcloud/api"
+	"github.com/canonical/microcloud/microcloud/api/types"
+	"github.com/canonical/microcloud/microcloud/client"
+	"github.com/canonical/microcloud/microcloud/cmd/style"
+	"github.com/canonical/microcloud/microcloud/service"
+)
+
+// Warning represents a warning message with a severity level.
+type Warning struct {
+	Level   StatusLevel
+	Message string
+}
+
+// Warnings is a list of warnings.
+type Warnings []Warning
+
+// Status returns the overall status of the warning list.
+// If there are any Error level warnings, the status will be error.
+// Otherwise, if there are any Warn level warnings, the status will be warn.
+// Finally, the status will be Success, implying no warnings.
+func (w Warnings) Status() StatusLevel {
+	if len(w) == 0 {
+		return Success
+	}
+
+	for _, warning := range w {
+		if warning.Level == Error {
+			return Error
+		}
+	}
+
+	return Warn
+}
+
+// StatusLevel represents the severity level of warnings.
+type StatusLevel int
+
+const (
+	// Success represents a lack of warnings.
+	Success StatusLevel = iota
+
+	// Warn represents a medium severity warning.
+	Warn
+
+	// Error represents a critical warning.
+	Error
+)
+
+// Symbol returns the single-character symbol representing the StatusLevel, color coded.
+func (s StatusLevel) Symbol() string {
+	switch s {
+	case Success:
+		return style.SuccessSymbol()
+	case Warn:
+		return style.WarningSymbol()
+	case Error:
+		return style.ErrorSymbol()
+	}
+
+	return ""
+}
+
+// Symbol returns a word representing the StatusLevel, color coded.
+func (s StatusLevel) String() string {
+	switch s {
+	case Success:
+		return style.SuccessColor("HEALTHY")
+	case Warn:
+		return style.WarningColor("WARNING")
+	case Error:
+		return style.ErrorColor("ERROR")
+	}
+
+	return ""
+}
+
+type cmdStatus struct {
+	common *CmdControl
+}
+
+func (c *cmdStatus) Command() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "status",
+		Short: "Deployment status with configuration warnings",
+		RunE:  c.Run,
+	}
+
+	return cmd
+}
+
+func (c *cmdStatus) Run(cmd *cobra.Command, args []string) error {
+	if len(args) != 0 {
+		return cmd.Help()
+	}
+
+	cloudApp, err := microcluster.App(microcluster.Args{StateDir: c.common.FlagMicroCloudDir})
+	if err != nil {
+		return err
+	}
+
+	status, err := cloudApp.Status(context.Background())
+	if err != nil {
+		return fmt.Errorf("Failed to get MicroCloud status: %w", err)
+	}
+
+	if !status.Ready {
+		return fmt.Errorf("MicroCloud is uninitialized, run 'microcloud init' first")
```

Comment:
In the [first commit](https://github.com/canonical/microcloud/pull/384/commits/71ffd29b8cbdbbca1f67d889e8f4477b13f3485b#diff-1f9d01a7137adb7eda731cd49f4e87908c5f90f56beac88c93792ea7cf504ad3R30) I saw the status endpoint is enabled in pre-init mode. Should we either remove this check or don't let the endpoint be available before init?

Reply:
It shouldn't be available prior to init, so I've removed the flag on the handler.